agent/fw_drupal.c

```diff
@@ -67,6 +67,90 @@ NR_PHP_WRAPPER(nr_drupal_qdrupal_name_the_wt) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_7_3_X_API_NO
+static void nr_drupal_http_request_ensure_second_arg(NR_EXECUTE_PROTO) {
+  zval* arg = NULL;
+
+  /*
+   * If only one argument is given, an empty list is inserted as second
+   * argument. NR headers are added during a later step.
+   */
+  if (ZEND_NUM_ARGS() == 1) {
+    arg = nr_php_zval_alloc();
+    array_init(arg);
+
+    nr_php_arg_add(NR_EXECUTE_ORIG_ARGS, arg);
+
+    nr_php_zval_free(&arg);
+  }
+}
+
+static zval* nr_drupal_http_request_add_headers(NR_EXECUTE_PROTO TSRMLS_DC) {
+  bool is_drupal_7;
+  zval* second_arg = NULL;
+  zend_execute_data* ex = nr_get_zend_execute_data(NR_EXECUTE_ORIG_ARGS);
+
+  if (nrunlikely(NULL == ex)) {
+    return NULL;
+  }
+
+  /* Ensure second argument exists in the call frame */
+  nr_drupal_http_request_ensure_second_arg(NR_EXECUTE_ORIG_ARGS);
+
+  is_drupal_7 = (ex->func->common.num_args == 2);
+
+  /*
+   * nr_php_get_user_func_arg is used, as nr_php_arg_get calls ZVAL_DUP
+   * on the argument zval and thus doesn't allow us to change the
+   * original argument.
+   */
+  second_arg = nr_php_get_user_func_arg(2, NR_EXECUTE_ORIG_ARGS);
```

Comment:
Could check if this value is `NULL` and return immediately without having to call `nr_drupal_headers_add`

Reply:
I didn't change the original implementation which didn't check for `NULL` before calling `nr_drupal_headers_add`. This was pure copy/paste re-factoring to re-use code in pre-oapi and oapi instrumentation. It is highly unlikely that `second_arg` will be `NULL` so I think for code clarity this can be left as is.